src/parsing/astHelper.ts

```diff
@@ -171,6 +171,10 @@ function followPath(ast: IAST, path: string[]): IAST | null {
 /**
  * Get the value of the given attribute
  */
+function getAttribute(
+	ast: IAST,
+	attributeName: 'type' | Omit<string, 'type'>
+): string | SequenceType | null;
```

Comment:
Since you take attributeName: 'type' here, can't you just return SequenceType | null? I think that can also simplify some casts later in the code (as SequenceType).

Reply:
Good point, but it seemed to break some other things.

Reply:
Now it works.

Reply:
Wonderful!